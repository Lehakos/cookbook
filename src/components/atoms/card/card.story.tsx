import React from 'react';
import { storiesOf } from '@storybook/react';

import { Card } from './card';

const stories = storiesOf('Atoms|Card', module).addParameters({
  component: Card,
});

stories.add('Default', () => (
  <Card py={4} px={5}>
    Lorem ipsum, dolor sit amet consectetur adipisicing elit. Officiis
    accusantium dolore ab dolorem totam, molestiae veniam possimus officia, esse
    velit similique sed amet voluptatem. Doloremque fugiat dolorum quidem libero
    odio?
  </Card>
));
