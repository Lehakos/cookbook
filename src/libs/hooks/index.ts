export { useControlledProp } from './use-controlled-prop';
export { useElementSize } from './use-element-size';
