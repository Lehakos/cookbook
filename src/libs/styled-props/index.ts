export { typography, TypographyProps } from './typography';
export { layout, LayoutProps } from './layout';
export { flexbox, FlexboxProps } from './flexbox';
