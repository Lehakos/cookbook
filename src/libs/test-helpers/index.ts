export { render } from './render';
export { mockElementWithBoundingRect } from './mock-element-with-bounding-rect';
