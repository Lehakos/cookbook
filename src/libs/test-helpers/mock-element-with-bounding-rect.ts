export const mockElementWithBoundingRect = ({
  width = 100,
  height = 100,
  top = 0,
  left = 0,
}) => {
  const right = left + width;
  const bottom = top + height;

  Object.defineProperties(Element.prototype, {
    offsetLeft: {
      get() {
        return left;
      },
      configurable: true,
    },
    offsetTop: {
      get() {
        return top;
      },
      configurable: true,
    },
    offsetHeight: {
      get() {
        return height;
      },
      configurable: true,
    },
    offsetWidth: {
      get() {
        return width;
      },
      configurable: true,
    },
  });

  Element.prototype.getBoundingClientRect = jest.fn(() => {
    return {
      width,
      height,
      top,
      left,
      bottom,
      right,
    };
  });

  const element = document.createElement('div');

  return element;
};
