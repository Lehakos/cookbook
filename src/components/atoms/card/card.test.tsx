import React from 'react';

import { render } from 'libs/test-helpers';

import { Card } from './card';

describe('<Card />', () => {
  it('renders properly', () => {
    const { container } = render(<Card />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
