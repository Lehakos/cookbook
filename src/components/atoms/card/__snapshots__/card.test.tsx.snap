// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> renders properly 1`] = `
.c0 {
  padding: 12px 16px;
  border-radius: 4px;
  box-shadow: 0 0 4px 0 rgba(0,0,0,.2);
  background: #fff;
}

<div
  class="c0"
/>
`;
